docroot/modules/custom/annotation_store/src/Controller/AnnotationStoreController.php

```diff
@@ -0,0 +1,192 @@
+      $split = explode('/', $path_alias['source']);
+      $resource_entity_id = $split[2];
+    }
+    // If url is without alias.
+    else {
+      $split = explode('/', $request_path);
+      $resource_entity_id = $split[2];
+    }
+    return $resource_entity_id;
+  }
+
+  /**
+   * Annotation search - Returns list of annotations.
+   */
+  public function annotationApiSearch() {
+    $ids = \Drupal::entityQuery('annotation_store')->condition('resource_entity_id', $this->getResourceEntityId())->execute();
+    $obj = entity_load_multiple('annotation_store', $ids);
+    $res = '';
+    if ($obj) {
+      foreach ($obj as $annotations) {
+        $res .= $annotations->data->value;
+        if ($annotations !== end($obj)) {
+          $res .= ',';
+
+        }
+      }
+      $res = rtrim($res, ",");
+    }
+    else {
+      // Dummy data for player initialization when data is absent.
+      // Temporary Fix - for Open Video Annotation.
+      $res = '{"permissions":{"read":[],"update":[],"delete":[],"admin":[]},"ranges":[],"quote":"","text":"dummy","media":"video","target":{"container":"vjs_video_dummy","src":"http:\/\/dummy.com\/dummy.mp4","ext":".mp4"},"rangeTime":{"start":3.14796,"end":4.65196},"updated":"2016-07-08T07:23:10.147Z","created":"2016-07-08T07:23:10.147Z","uri":"http:\/\/dummy.com\/dummy\/1","id":"1"}';
+    }
+    $ars = '{"rows":[' . $res . ']}';
+    print $ars;
+    exit;
+  }
+
+  /**
+   * Annotation create as entity.
+   */
+  public function annotationApiCreate() {
+    $annotation_data = $this->annotationApiFromStdin();
+    $annotation_data_save = $annotation_data;
+    $language = \Drupal::languageManager()->getCurrentLanguage()->getId();
+    if ($annotation_data->text) {
+      $entity = entity_create('annotation_store', array(
+        'type' => $annotation_data->media,
+        'language' => $language,
+        'data' => json_encode($annotation_data),
+        'uri' => $annotation_data->uri,
+        'text' => $annotation_data->text,
+        'resource_entity_id' => $this->getResourceEntityId(),
+      ));
+      $entity->save();
+      $annotation_data->id = $entity->id();
+      $this->updateAnnotation($entity->id(), $annotation_data, 'onCreate');
+      print_r(json_encode($annotation_data));
+    }
+    exit;
+  }
+
+  /**
+   * Annotation update - loads posted data, returns data as JSON object.
+   */
+  public function annotationApiUpdate($id) {
+    $annotation_data = $this->annotationApiFromStdin();
+    if ($id) {
+      $this->updateAnnotation($id, $annotation_data, 'onUpdate');
+      print_r(json_encode($annotation_data));
+    }
+    else {
+      print_r('failed');
+    }
+    exit;
+  }
+
+  /**
+   * Annotation update - deletes the entity based on the id passed.
+   */
+  public function annotationApiDelete() {
+    $data = $this->annotationApiFromStdin();
+    $id = $data->id;
+    if ($id) {
+      entity_delete_multiple('annotation_store', array($id));
+      print_r(1);
```

Comment:
Doesn't look right. Please use proper response class to respond back rather than print & exit.